lib/coinbase/collectMoment.ts

```diff
@@ -0,0 +1,90 @@
+import { Address, encodeFunctionData, erc20Abi, formatUnits, Hash, parseUnits } from "viem";
+import { z } from "zod";
+import { CHAIN_ID, IS_TESTNET, USDC_ADDRESS } from "@/lib/consts";
+import { sendUserOperation } from "@/lib/coinbase/sendUserOperation";
+import { erc20MinterABI } from "@zoralabs/protocol-deployments";
+import { getOrCreateSmartWallet } from "./getOrCreateSmartWallet";
+import { collectSchema } from "./collectSchema";
+import getTokenInfo from "../viem/getTokenInfo";
+import getBalance from "../viem/getBalance";
+import getAllowance from "../viem/getAllowance";
+import { erc20MinterAddresses } from "../protocolSdk/constants";
+
+export type CollectMomentInput = z.infer<typeof collectSchema>;
+
+export interface CollectResult {
+  hash: Hash;
+  chainId: number;
+}
+
+/**
+ * Collect a Zora 1155 token using a smart account via Coinbase CDP.
```

Comment:
optional: avoid using `zora` 
- instead, use `In Process`

```suggestion
 * Collect an In Process 1155 token using a smart account via Coinbase CDP.
```